core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java

```diff
@@ -183,6 +187,21 @@ public BigQuerySqlDialect(SqlDialect.Context context) {
     case TRIM:
       unparseTrim(writer, call, leftPrec, rightPrec);
       break;
+    case ITEM:
+      if (call.operand(0) instanceof SqlBasicCall
```

Comment:
I tried to handle the SQL which access the element of a table column like
```sql
WITH POST.DEPT_GBY_EMP_NAME AS {
      select * from (values
              (10, array['Jane', 'Bob']),
              (20, array['Eric']),
              (30, array['Susan', 'Alice']),
              (40, array[null])) as t(deptno, employees)
}
SELECT D.EMPLOYEES[1] FROM POST.DEPT_GBY_EMP_NAME D
```
I found it's hard to recognize which type the column is here. It could be a `ARRAY`, `MAP` or `STRUCT`.  Is there any better way to do this here?

Reply:
Sorry that I didn't realized that in `SqlDialect` we do not have validated types. It's indeed not easy to know the operand's type here.
I would suggest that we go back to the JIRA to discuss the design first, I'll post my concerns there.